Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py

```diff
@@ -229,22 +232,37 @@ def init_roots(self):
             # disable logging as we might receive client error and try 2.1
             logging.disable(logging.ERROR)
             # try TAXII 2.0
-            self.api_root = self.server.api_roots[0]  # type: ignore[union-attr, attr-defined]
-            # override _conn - api_root isn't initialized with the right _conn
-            self.api_root._conn = self._conn  # type: ignore[attr-defined]
+            self.set_api_root()
         # (TAXIIServiceException, HTTPError) should suffice, but sometimes it raises another type of HTTPError
         except Exception as e:
             if "406 Client Error" not in str(e):
                 raise e
             # switch to TAXII 2.1
             self.init_server(version=TAXII_VER_2_1)
-            self.api_root = self.server.api_roots[0]  # type: ignore[union-attr, attr-defined]
-            # override _conn - api_root isn't initialized with the right _conn
-            self.api_root._conn = self._conn  # type: ignore[attr-defined]
+            self.set_api_root()
         finally:
             # enable logging
             logging.disable(logging.NOTSET)
 
+    def set_api_root(self):
+        roots_to_api = {str(api_root.url).split('/')[-2]: api_root
+                        for api_root in self.server.api_roots}  # type: ignore[union-attr, attr-defined]
+
+        if self.default_api_root:
+            if not roots_to_api.get(self.default_api_root):
+                raise DemistoException(f'The given default API root {self.default_api_root} doesn\'t exists.'
+                                       f'Available API roots are {list(roots_to_api.keys())}.')
+            self.api_root = roots_to_api.get(self.default_api_root)  # type: ignore[union-attr, attr-defined]
+
+        elif self.server.default:  # type: ignore[union-attr, attr-defined]
+            self.api_root = self.server.default  # type: ignore[union-attr, attr-defined]
+
+        else:
+            self.api_root = self.server.api_roots[0]  # type: ignore[union-attr, attr-defined]
+
+        # override _conn - api_root isn't initialized with the right _conn
+        self.api_root._conn = self._conn  # type: ignore[union-attr, attr-defined]
```

Comment:
Is it possible to deal with these types of errors instead of ignoring them?

Reply:
Updated and limited ignores to the only ones needed.